src/libutil/thread.cpp

```diff
@@ -49,15 +49,72 @@
 #include <future>
 #include <memory>
 
+#include <OpenImageIO/sysutil.h>
+#include <OpenImageIO/thread.h>
+#include <OpenImageIO/sysutil.h>
+#include <OpenImageIO/strutil.h>
+
+#include <boost/thread/tss.hpp>
+
+#if 0
+
 // Use boost::lockfree::queue for the task queue
 #include <boost/lockfree/queue.hpp>
+template<typename T> using Queue = boost::lockfree::queue<T>;
 
-#include <OpenImageIO/sysutil.h>
-#include <OpenImageIO/thread.h>
+#else
+
+#include <queue>
+
+namespace {
+
+template <typename T>
+class Queue {
+public:
+    Queue (int size) { }
+    bool push(T const & value) {
+        std::unique_lock<Mutex> lock(this->mutex);
+        this->q.push(value);
+        return true;
+    }
+    // deletes the retrieved element, do not use for non integral types
+    bool pop(T & v) {
+        std::unique_lock<Mutex> lock(this->mutex);
+        if (this->q.empty())
+            return false;
+        v = this->q.front();
+        this->q.pop();
+        return true;
+    }
+    bool empty() {
+        std::unique_lock<Mutex> lock(this->mutex);
+        return this->q.empty();
+    }
+private:
+    typedef OIIO::spin_mutex Mutex;
+    std::queue<T> q;
+    Mutex mutex;
+};
+
+}
+
+#endif
 
 
 OIIO_NAMESPACE_BEGIN
 
+bool debug_threads = false; // FIXME -- remove this after debugging
+
+static int
+threads_default ()
+{
+    int n = Strutil::from_string<int>(Sysutil::getenv("OPENIMAGEIO_THREADS"));
+    if (n < 1)
```

Comment:
I think so -- just getting a little confused over which was which in which context. :-)